.github/workflows/test_template.yaml

```
@@ -1,20 +1,30 @@
-name: Test
-on: [push, pull_request]
+name: Workflow Test Template
+on: 
+  workflow_call:
+    inputs:
+      runs-on:
+        required: false
+        type: string
+        default: ubuntu-latest
+      targets:
+        required: false
+        type: string
+        default: "['linux-unit-test-1-cpu', 'linux-unit-test-4-cpu']"
+permissions: read-all
+
 jobs:
   run:
-    runs-on: ubuntu-latest
+    runs-on: ${{ inputs.runs-on }}
     strategy:
       fail-fast: false
       matrix:
-        target:
-          - linux-amd64-unit-4-cpu-race
-          - linux-386-unit-1-cpu
+        target: ${{ fromJSON(inputs.targets) }}
     steps:
       - uses: actions/checkout@v4
       - uses: actions/setup-go@v5
         with:
           go-version: "1.21.6"
-      - env:
+      - env: 
           TARGET: ${{ matrix.target }}
         run: |
           case "${TARGET}" in
```

Looks like the `arm64` test is showing unexpected caching behaviour: https://github.com/etcd-io/raft/actions/runs/7589241324/job/20673470700?pr=128 

> ```bash
> Starting at: Fri Jan 19 21:14:04 UTC 2024
> 'unit' started at Fri Jan 19 21:14:04 UTC 2024
> ok  	go.etcd.io/raft/v3	2.651s
> ok  	go.etcd.io/raft/v3/confchange	1.468s
> ok  	go.etcd.io/raft/v3/quorum	2.166s
> ok  	go.etcd.io/raft/v3/raftpb	(cached)
> ok  	go.etcd.io/raft/v3/rafttest	(cached)
> ok  	go.etcd.io/raft/v3/tracker	(cached)
> 'unit' completed at Fri Jan 19 21:14:07 UTC 2024
> SUCCESS
> ```


Let's prevent that as follows:

```suggestion
          go clean -testcache
          case "${TARGET}" in
```